grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -283,6 +286,101 @@ class StockMovementController {
+        render(template: "comments", model: [stockMovement: stockMovement])
+    }
+
+    def addComment = {
+        StockMovement stockMovement = stockMovementService.getStockMovement(params.id)
+        [stockMovement: stockMovement, comment: new Comment()]
+    }
+
+    def editComment = {
+        StockMovement stockMovement = stockMovementService.getStockMovement(params['stockMovement.id'])
+        if (!stockMovement) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'stockMovement.label', default: 'Stock Movement'), params['stockMovement.id']])}"
+            redirect(action: "list")
+            return
+        }
+        Comment comment = Comment.get(params?.id)
+        if (!comment) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'comment.label', default: 'Comment'), comment.id])}"
+            redirect(action: "show", id: stockMovement?.id)
+            return
+        }
+        if (comment.sender.id != session.user.id) {
+            flash.message = "${g.message(code: 'auth.notAuthorized.message')}"
+            redirect(action: "show", id: stockMovement?.id)
+            return
+        }
+        render(view: "addComment", model: [stockMovement: stockMovement, comment: comment])
+    }
+
+    def deleteComment = {
+        StockMovement stockMovement = stockMovementService.getStockMovement(params['stockMovement.id'])
+        if (!stockMovement) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'stockMovement.label', default: 'StockMovement'), params['stockMovement.id']])}"
+            redirect(action: "list")
+            return
+        }
+        Comment comment = Comment.get(params?.id)
+        if (!comment) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'comment.label', default: 'Comment'), params.id])}"
+            redirect(action: "show", id: stockMovement.id)
+            return
+        }
+
+        if (comment.sender.id != session.user.id) {
+            throw new UnsupportedOperationException("${g.message(code: 'errors.noPermissions.label')}")
+        }
+        stockMovementService.deleteComment(comment, stockMovement.requisition)
+        flash.message = "${g.message(code: 'default.comments.deleted.label')}"
+        redirect(action: "show", id: stockMovement.id)
+    }
+
+    def saveComment = {
+        StockMovement stockMovement = stockMovementService.getStockMovement(params['stockMovement.id'])
+        if (!stockMovement) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'stockMovement.label', default: 'StockMovement'), params['stockMovement.id']])}"
+            redirect(action: "list")
+            return
+        }
+        Comment comment = new Comment(params)
+        stockMovementService.saveComment(comment)
```

Comment:
it is not needed, because the comment would be save by cascade in the method below `addCommentToRequisition`.

Reply:
I need to get the validation errors, so it's either I save it and get the error bean or do a validate on a comment inside addCommentToRequisition()
cascade save requisition did not return any errors fro me.

Reply:
I knew that. This is why I suggested the `comment.validate()` in the comment below